### ISSUE TITLE
Pronote KO

### ISSUE THREAD

**Issue by NuoLeonis** [delphiki/hass-pronote#93]
Depuis la dernière mise à jour de HA erreur  de l'intégration PRONOTE :

'Period' object has no attribute '_client' and no __dict__ for setting new attributes

HA
2024.12
Core
24.11.3

**Comment by delphiki** [comment]
Hello, souci connu en effet, une PR est ouverte côté pronotepy : https://github.com/bain3/pronotepy/pull/317 👍

**Comment by ernatono** [comment]
Hello. Une maj est prévue ? Merci.

**Comment by delphiki** [comment]
Hello,
Je ferai la màj de l'intégration dès que les correctifs seront intégrés dans pronotepy 😉

**Comment by ernatono** [comment]
ça marche merci !

----- Mail original -----
De: "ju" ***@***.***>
À: "delphiki/hass-pronote" ***@***.***>
Cc: "ernatono" ***@***.***>, "Comment" ***@***.***>
Envoyé: Jeudi 5 Décembre 2024 11:52:53
Objet: Re: [delphiki/hass-pronote] Pronote KO (Issue #93)





Hello, 
Je ferai la màj de l'intégration dès que les correctifs seront intégrés dans pronotepy 😉

**Comment by delphiki** [comment]
Hello @Sami6053,

Je n'ai pas de délai de prise en compte pour pronotepy (logique, je ne donne pas plus de délais pour les prises en charge des modifs qu'on me soumet :)), mais j'ai un fix temporaire qui fonctionne sur ma version (en utilisant mon fork de pronotepy), donc je vais publier un correctif ce soir en version bêta 👍

**Comment by Sami6053** [comment]
ok @delphiki ,
merci pour ton temps et tes efforts

**Comment by delphiki** [comment]
Je viens de publier la version 1.15.1-dev qui devrait résoudre le souci.

**Comment by dudu7731** [comment]
je confirme que cette version marche :)

**Comment by tdelaux** [comment]
Bonjour,
Possible de la mettre pour une installation via HACS ?

**Comment by tdelaux** [comment]
C'est bon, j'ai trouvé...
Pour les ceux qui cherchent comme moi dans Hacs ==> les 3 petits point ==> Retélécharger

**Comment by kornesque** [comment]
Oui, il faut cliquer sur les 3 petit points à droite, puis re-download et enfin choisir une autre version.
Là tu choisi la version 1.15.1-dev

**Comment by Sami6053** [comment]
c'est parfait
merci @delphiki

**Comment by rafal83** [comment]
> Hello, Je ferai la màj de l'intégration dès que les correctifs seront intégrés dans pronotepy 😉

Sur pronotepy, il indique que le projet est passé en maintenance, donc plus aucune PR ne sera acceptée.

**Comment by delphiki** [comment]
> > Hello, Je ferai la màj de l'intégration dès que les correctifs seront intégrés dans pronotepy 😉
> 
> Sur pronotepy, il indique que le projet est passé en maintenance, donc plus aucune PR ne sera acceptée.

Plus aucune PR concernant des nouvelles features, mais les correctifs et adaptations pour que la lib fonctionne avec les mises à jour de Pronote oui. :)

**Comment by MerlinHACS** [comment]
Bonjour à tous,
j'ai bien mis la version 1.15.1-dev et restart HA, malgré cela mes capteurs pronote sont toujours en "Etat" "Non fournies" (point d'exclamation rouge dans un cercle flèche).
je n'ose pas effacer les capteurs car j'ai peur de ne plus savoir comment j'ai fait pour avoir tout le pronote de mes enfants :-(

**Comment by skillix** [comment]
> Bonjour à tous, j'ai bien mis la version 1.15.1-dev et restart HA, malgré cela mes capteurs pronote sont toujours en "Etat" "Non fournies" (point d'exclamation rouge dans un cercle flèche). je n'ose pas effacer les capteurs car j'ai peur de ne plus savoir comment j'ai fait pour avoir tout le pronote de mes enfants :-(

J'ai eu le soucis également, mais je viens de trouver le PB, Tu dois probablement comme moi passer par le QRCODE, mais le format a changer, il y a un champs en plus dans le JSON, il faut supprimer : `"avecPageConnexion":false,` au tout début.

![image](https://github.com/user-attachments/assets/b6031f42-3738-43ad-a7f0-18abc632e3c5)

Supprime cette partie et ca fonctionne

**Comment by ernatono** [comment]
Bonjour et meilleurs voeux !

J'ai un soucis sur un des 2 comptes pronote avec ce message alors que tout marchait bien :

Échec de la configuration, nouvel essai: ('Decryption failed while trying to un pad. (probably bad decryption key/iv)', 'exception happened during login -> probably bad username/password')

Une idée ?

Merci.


----- Mail original -----
De: "skillix" ***@***.***>
À: "delphiki/hass-pronote" ***@***.***>
Cc: "ernatono" ***@***.***>, "Comment" ***@***.***>
Envoyé: Mardi 7 Janvier 2025 09:05:06
Objet: Re: [delphiki/hass-pronote] Pronote KO (Issue #93)







Bonjour à tous, j'ai bien mis la version 1.15.1-dev et restart HA, malgré cela mes capteurs pronote sont toujours en "Etat" "Non fournies" (point d'exclamation rouge dans un cercle flèche). je n'ose pas effacer les capteurs car j'ai peur de ne plus savoir comment j'ai fait pour avoir tout le pronote de mes enfants :-( 

J'ai eu le soucis également, mais je viens de trouver le PB, Tu dois probablement comme moi passer par le QRCODE, mais le format a changer, il y a un champs en plus dans le JSON, il faut supprimer : "avecPageConnexion":false, au tout début. 

image.png (view on web) 

Supprime cette partie et ca fonctionne

**Comment by skillix** [comment]
j'ai eu un soucis hier, impossible de refaire le Qrcode, avec un message d'erreur inconnu, j'ai regardé, et repasser sur le derniere version : 

![image](https://github.com/user-attachments/assets/d1338b88-88d6-4243-81d6-7d3e91b3acbd)

Passage du json (toujours sans le paramètre en trop) et tout est reparti. (suppression et recréation d'une nouvelle entrée)
